submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+| `--kubecontext` `<string>` | Kubeconfig context to use
+| `--name` `<string>`        | The name of the policy to export
+
+##### `list`
+
+`subctl connection-policy list [flags]`
+
+Inspects the cluster and reports information about the detected connection policies.
+
+| Flag                         | Description
+|:-----------------------------|:----------------------------------------------------------------------------|
+| `--kubeconfig` `<string>`    | Absolute path(s) to the kubeconfig file(s) (default `$HOME/.kube/config`)
+| `--kubecontext` `<string>`   | Kubeconfig context to use
+
+#### Connection Policy Examples
+
+- we want to specify a default connection policy that uses a full-mesh topology with IPsec as the default cable driver.
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: public-internet
+spec:
+  connectionPolicy: connect
+  topology: full-mesh
+  cableDriver:
+    name: "ipsec"
+  priority: 0
+```
+
+- we want to specify a default connection policy that uses a doesn't connect clusters.
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: public-internet
+spec:
+  connectionPolicy: no-connect
+  priority: 0
+```
+
+- We want to use specific cable drivers to connect clusters in a cluster set as shown in the image below. We
+label the `Clusters` appropriately and define a number of policies with the appropriate label selectors.
+
+![Finer grained connectivity example](./images/multiple_cables_ep.png)
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: on-prem
+spec:
+  leftClusterSelector:
+    location: on-premise
+  rightClusterSelector:
+    location: on-premise
+  topology: point-to-point
+  connectionPolicy: connect
+  cableDriver:
+    name:  "vxlan"
+  priority: 3
+```
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: on-prem-to-cloud
+spec:
+  leftClusterSelector:
+    location: on-premise
+    env: production
+  rightClusterSelector:
+    location: cloud
+    env: production
+  topology: point-to-point
+  connectionPolicy: connect
+  cableDriver:
+    name:  "ipsec"
+  priority: 3
+```
+
+> **_NOTE:_** we could have just used the `env: production` label in the example above.
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: cloud-a-default
+spec:
+  clusterSelector:
+    location: cloud
+  topology: full-mesh
+  cableDriver:
+    name:  "vxlan"
+  priority: 3
+```
+
+- We don't want to connect two clusters in a `ClusterSet`. We label the `Clusters` appropriately and define a policy with the appropriate
+label selectors.
+
+```yaml
+kind: ClusterConnectionPolicies
+metadata:
+  name: on-prem-to-cloud-no-connect
+spec:
+  leftClusterSelector:
+    location: on-premise
+    env: production
+  rightClusterSelector:
+    location: cloud
+    env: production
+  connectionPolicy: no-connect
+  priority: 3
+```
+
+- We only want certain clusters to access a database server in another cluster and no other clusters to connect to one another. We
```

Comment:
This use case matches "Multi Cluster Network Policies" and should be covered by that effort, not by limiting connectivity as a whole

Reply:
I guess in one sense it is but the GWs will have no awareness of what they can/can't connect to as with the multicluster network policy everything is handled at an external level - so there could be wasted resources on the GW side... I would need to look at this more closely